src/apps/SafeConnect/App.tsx

```diff
@@ -0,0 +1,211 @@
+          if (code && code.data) {
+            setIsConnecting(true);
+            wcConnect(code.data);
+          } else {
+            setInvalidQRCode(true);
+            setWcUri(
+              `Screen Shot ${format(new Date(), "yyyy-MM-dd")} at ${format(
+                new Date(),
+                "hh.mm.ss"
+              )}`
+            );
+          }
+        };
+        reader.readAsDataURL(blob);
+      }
+    },
+    [wcClientData, wcConnect]
+  );
+
+  const getDisconnectedContent = () => {
+    if (isConnecting) {
+      return <Loader size="md" />;
+    }
+    return (
+      <TextField
+        id="wc-uri"
+        label="WalletConnect URI"
+        value={wcUri}
+        onChange={(e) => setWcUri(e.target.value)}
+        onPaste={onPaste}
+        meta={{
+          error: invalidQRCode ? "Invalid QR code" : undefined,
+        }}
+      />
+    );
+  };
+
+  const getConnectedContent = () => (
+    <>
+      <WCClientInfo
+        name={wcClientData!.name}
+        url={wcClientData!.url}
+        iconSrc={wcClientData!.icons[0]}
+      />
+
+      <Button
+        size="md"
+        color="primary"
+        variant="contained"
+        onClick={() => wcDisconnect()}
+      >
+        Disconnect
+      </Button>
+    </>
+  );
+
+  const getConnectedInstructions = () => (
+    <>
+      <StyledTitle size="sm">How to connect to a Dapp</StyledTitle>
+
+      <StyledText size="lg">
+        1) Open a Dapp with WalletConnect support.
+      </StyledText>
+
+      <StyledText size="lg">
+        2) Copy QR code image into clipboard (Shift+Command+4 on Mac, Windows
+        key+PrtScn on Windows).
+      </StyledText>
+
+      <StyledText size="lg">
+        3) Past QR code image into the input field (Command+V on Mac, Ctrl+V on
+        Windows)
+      </StyledText>
+
+      <StyledText size="lg">
+        4) WalletConnect connection will be established automatically.
+      </StyledText>
+
+      <StyledText size="lg">
+        5) Now you can trigger transactions view the Dapp to your Safe.
+      </StyledText>
+
+      <HelpLink />
+    </>
+  );
+
+  const getDisconnectedInstructions = () => (
+    <>
+      <StyledTitle size="sm">How to confirm transactions</StyledTitle>
+
+      <StyledText size="lg">1) Trigger a transaction from the Dapp.</StyledText>
+
+      <StyledText size="lg">
+        2) Come back here to confirm the transaction. You will see a popup with
+        transactions details. Review the details and submit the transaction.
+      </StyledText>
+
+      <StyledText size="lg">
+        3) The transaction has to be confirmed be owners and executed just like
+        any other Safe transaction.
+      </StyledText>
+
+      <HelpLink />
+    </>
+  );
+
+  useEffect(() => {
+    let interval: number;
+    if (isConnecting) {
+      interval = setTimeout(() => setIsConnecting(false), 2_000);
+    }
+    return () => clearInterval(interval);
```

Comment:
`clearInterval` is used with `setInterval` shouldn't we use `clearTimeout` here?